docs/brexit_checker.md

```diff
@@ -0,0 +1,109 @@
+
+### Question
+A question represents a question presented to the user and the possible answers they can respond with. Each option respresents a criterion to apply to the user. For example, the question "Do you own a business?" may have the options "Yes" and "No". If the user responded "Yes", they would be attributed with the criterion "owns-a-business". 
+
+| attribute   | values                                                            | description                                                        | required |
+|-------------|-------------------------------------------------------------------|--------------------------------------------------------------------|----------|
+| key         | string                                                            | unique id                                                          | yes      |
+| type        | either 'single', 'single_wrapped', 'multiple', 'multiple_grouped' | determines the style and behaviour of the question                 | yes      |
+| text        | string                                                            | question text                                                      | yes      |
+| hint_text   | string                                                            | description on how make a answer selection                         | no       |
+| description | string                                                            | additional context on the question being asked                     | no       |
+| options     | array                                                             | list of possible answers                                           | yes      |
+| criteria    | Criteria Rule                                                     | defines for which set of criteria the question should be presented | no       |
+
+Question can be conditionally shown depending on the user's current list of criteria. This allows us to support different question flows depending on a users circumstances e.g. we don't show questions about businesses if they answered they don't own a business.
+
+### Criteria Rule
+A Criteria Rule is an object which defines a set of criteria that is applicable.
+
+| attribute | values                                          | description                                                  | required                 |
+|-----------|-------------------------------------------------|--------------------------------------------------------------|--------------------------|
+| all_of    | list containing criteria keys or Criteria Rules | is true when all element of the list is present or also true | if any_of is not present |
+| any_of    | list containing criteria keys or Criteria Rules | is true when any element of the list is present or also true | if all_of is not present |
+
+Criteria Rule is expressed using the `all_of` or `any_of` keys. The value of these keys is a list of criteria keys, nested Criteria Rules or both. For example:
```

Comment:
Since we've already said what the values of `all_of` and `any_of` are in the table above, I don't think we need to repeat them here.